allocate.c

```diff
@@ -0,0 +1,12 @@
+#include "include/common.h"
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+	size_t size = 31;
+	printf("Size: ");
+	scanf("%lu", &size);
```

Comment:
For a small example program, I think it's probably OK not to check.

Reply:
I get that we're not processing stock market trades, but I have been known to loose entire days debugging single character typos before :). 

As small as it may be, it  is also a perfect example of how errors compound in the land of C.  Either a safe initialization value for `size` or a branch based off the return code of `scanf` would save you from the typo scenario above. Both would countermeasures would better and give redundancy for future code edits.

Reply:
I think for the small example programs here, some shortcuts are acceptable: there is no pretence that this is production code. Of course, we don't want to let active *bugs* in!

Reply:
Though I do agree that `scanf` is a bit easy for a user to get wrong so something simple like:

```c
if (scanf(...) < 1)
    err(...);
```

is probably a reasonable compromise.

Reply:
Fixed like that: https://github.com/capablevms/cheri-examples/blob/6332f5d4b12ba8ded2e0f687d4e52f748b6ae534/set_bounds.c#L17-L19

Reply:
Works for me!